### PR TITLE
EIP1-10370-communication-count-stat

### PR DESCRIPTION
**Pull request by Callum-Griffiths**
## Describe your changes
Provide statistic for number of signature/identity requests made

## Issue ticket number and link

https://dluhcdigital.atlassian.net/browse/EIP1-10370

### CODE REVIEW

File: src/main/resources/openapi/NotificationsAPIs.yaml [hunk 1]
```diff
@@ -2327,11 +2330,17 @@ components:
           type: integer
         hasSentNotRegisteredToVoteCommunication:
           type: boolean
+        numSignatureRequestCommsSent:
```
Comment (DanWilkinsSoftwire) [inline src/main/resources/openapi/NotificationsAPIs.yaml:2333]: In the above properties, Communications is written out in full, which seems inconsistent with the new names. Not sure whether a change would be better or worse, as the properties are quite long as is, but thought I'd mention

**Review by DanWilkinsSoftwire — COMMENTED** [review]
I've had a look and had a couple of questions, however I don't know the codebase at all so I'd suggest getting someone else to take a look too!

File: src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt [hunk 2]
```diff
@@ -28,22 +28,23 @@ class StatisticsController(
...
 
         val hasSentNotRegisteredToVoteCommunication = notifications.any { it.type == NotificationType.NOT_REGISTERED_TO_VOTE }
 
         return CommunicationsStatisticsResponseVAC(
             photoRequested = photoRequested,
-            identityDocumentsRequested = identityDocumentsRequested,
+            identityDocumentsRequested = identityDocumentsRequested > 0,
             bespokeCommunicationsSent = bespokeCommunications,
             hasSentNotRegisteredToVoteCommunication = hasSentNotRegisteredToVoteCommunication,
+            numIdentityDocumentRequestCommsSent = identityDocumentsRequested,
```
[the remark this answers is not in the document]
Comment (Callum-Griffiths) [reply src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt:47]: You may just need to clean and rebuild the project

File: src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt [hunk 3]
```diff
@@ -57,30 +58,32 @@ class StatisticsController(
...
         val hasSentNotRegisteredToVoteCommunication = notifications.any { it.type == NotificationType.NOT_REGISTERED_TO_VOTE }
 
         return CommunicationsStatisticsResponseOAVA(
-            signatureRequested = signatureRequested,
-            identityDocumentsRequested = identityDocumentsRequested,
+            signatureRequested = signatureRequested > 0,
+            identityDocumentsRequested = identityDocumentsRequested > 0,
             bespokeCommunicationsSent = bespokeCommunications,
             hasSentNotRegisteredToVoteCommunication = hasSentNotRegisteredToVoteCommunication,
+            numSignatureRequestCommsSent = signatureRequested,
```
[the remark this answers is not in the document]
Comment (DanWilkinsSoftwire) [reply src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt:85]: Discussed

File: src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanWilkinsSoftwire) [reply src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt:47]: Discussed

**Review by AndrewYHuang — APPROVED** [review]
Looks good! I like the use of parameterized tests here too (though they might be a tad more thorough than strictly necessary)

**Pull request merged**